src-tauri/src/app.rs

```diff
@@ -143,62 +151,333 @@ impl Service {
-
-                    // Check if the event is associated with the currently selected calendar. We
-                    // only forward it up to the application if it is.
-                    //
-                    // @TODO: This is filtering out the operations _after_ they've been processed
-                    // by the backend. We should go through the whole processing pipeline when
-                    // selected and not at all when not.
-                    if let Some(selected_calendar) = state.selected_calendar {
-                        if selected_calendar != meta.calendar_id {
-                            debug!("ignoring stream event: calendar not selected");
-                            continue;
-                        };
-                        channel.send(ChannelEvent::Stream(event))?;
-                    }
+                    channel.send(ChannelEvent::Stream(event))?;
                 },
-                Some(event) = self.invite_codes_rx.recv() => {
-                    let json = match event {
-                        FromNetwork::GossipMessage { bytes, .. } => {
-                            serde_json::from_slice(&bytes)?
-                        },
-                        FromNetwork::SyncMessage { .. } => unreachable!(),
-                    };
-                    channel.send(ChannelEvent::InviteCodes(json))?;
-                }
+                // @TODO(sam): Need a way to handle ephemeral topics in the stream controller as
+                // we now don't have a static topic we can subscribe to on startup.
+                //
+                // Some(event) = self.invite_codes_rx.recv() => {
+                //     let json = match event {
+                //         FromNetwork::GossipMessage { bytes, .. } => {
+                //             serde_json::from_slice(&bytes)?
+                //         },
+                //         FromNetwork::SyncMessage { .. } => unreachable!(),
+                //     };
+                //     channel.send(ChannelEvent::InviteCodes(json))?;
+                // }
             }
         }
     }
 
-    async fn recv_channel(&mut self) -> anyhow::Result<Channel<ChannelEvent>> {
+    async fn recv_channel(&mut self) -> anyhow::Result<broadcast::Sender<ChannelEvent>> {
         let Some(channel) = self.channel_rx.recv().await else {
             return Err(anyhow::anyhow!("channel tx closed"));
         };
 
-        self.app_handle
-            .state::<Mutex<Context>>()
-            .lock()
-            .await
-            .channel_set = true;
+        self.context.lock().await.channel_set = true;
 
         Ok(channel)
     }
+}
 
-    fn spawn_invite_code_ready_watcher(&self, channel: Channel<ChannelEvent>) {
-        let rt = tokio::runtime::Handle::current();
-        let channel = channel.clone();
-        let invite_codes_ready = self.invite_codes_ready.clone();
-        rt.spawn(async move {
-            invite_codes_ready
+pub struct Rpc {
```

Comment:
This is the main RPC interface we're exposing to the frontend, here without any tauri dependencies, mainly so we can easily write tests, but also it's just nice to have it separate. 

Questionable naming in this module and it could do with splitting up into two modules probably.